src/app/pages/form-validation/form-validation.component.html

```diff
@@ -75,11 +85,22 @@ <h2>Password Validation</h2>
     </p>
     <mat-card>
         <form [formGroup]="userPassword">
-            <mat-form-field>
-                <input matInput placeholder="Old Password" required formControlName="oldPassword" type="password" />
+            <mat-form-field appearance="fill">
+                <mat-label>Old Password</mat-label>
+                <input
+                    matInput
+                    placeholder="Old Password"
+                    required
+                    formControlName="oldPassword"
+                    [type]="oldPasswordVisible ? 'text' : 'password'"
+                />
+                <button mat-icon-button matSuffix (click)="toggleOldPasswordVisibility()">
+                    <mat-icon>{{ oldPasswordVisible ? 'visibility' : 'visibility_off' }}</mat-icon>
+                </button>
             </mat-form-field>
             <div [formGroup]="passwordFormGroup">
-                <mat-form-field>
+                <mat-form-field appearance="fill">
```

Comment:
the toggle visibility button is not hooked up in the `type` attribute for the new/confirm password fields

Reply:
foolish oversight on my part...